master.dbo.sp_generate_merge.sql

```diff
@@ -513,75 +513,95 @@ END
 
 IF (@nologo = 0)
 BEGIN
- SET @output += @b + '--MERGE generated by ''sp_generate_merge'' stored procedure, Version 0.93'
+ SET @output += @b + '--MERGE/DELETE generated by ''sp_generate_merge'' stored procedure, Version 0.93'
  SET @output += @b + '--Originally by Vyas (http://vyaskn.tripod.com): sp_generate_inserts (build 22)'
  SET @output += @b + '--Adapted for SQL Server 2008/2012 by Daniel Nolan (http://danere.com)'
  SET @output += @b + ''
 END
 
-IF (@include_rowsaffected = 1) -- If the caller has elected not to include the "rows affected" section, let MERGE output the row count as it is executed.
- SET @output += @b + 'SET NOCOUNT ON'
- SET @output += @b + ''
+--All the hard work pays off here!!! You'll get your MERGE statement, when the next line executes!
+DECLARE @tab TABLE (ID INT NOT NULL PRIMARY KEY IDENTITY(1,1), val NVARCHAR(max));
+INSERT INTO @tab (val)
+EXEC (@Actual_Values)
 
+-- If no rows were returned, and we don't want to delete any rows, then terminate early
+IF ((SELECT COUNT(*) FROM @tab) <= 0 AND @delete_if_not_matched = 0)

```

Comment:
I note that a side effect of this PR is that, when the `@tab` table variable fails to populate (such as in the #23 scenario), that a `DELETE` statement is generated. Would it be possible to check for an error prior to the creation of the statement to ensure that a `DELETE` isn't accidentally scripted?